app/models/sources/joyent.rb

```diff
@@ -26,6 +26,24 @@ def validate_config!
       return false
     end
 
+    # Connnect to Joyent and find all of the machines with that combination of
+    # user and datacenter that is provided with each source object source.
+    # Each machine found will be delete.
+    def purge_servers(captcha=nil)
+      if captcha.nil?
+        puts("What is 1 + 2 + 3 + 4? Put answer as a parameter.")
+      elsif captcha == 10
+        connection.servers.select do |s|
```

Comment:
@MinutemanZ in the above code, in the proxies block, should those be `p` instead of `self`? Also:

1. Why soft delete the proxies instead of outright destroying them?
2. Shouldn't we instruct all sources to forget them?
3. If yes, does that also get rid of the performance record?